Support for real SSL certificates at the PZH Web Server.

jplyle opened the pull request:
http://jira.webinos.org/browse/WP-396

Step one in WP-396 to remove our dependency on self-signed certificates.  
This change is necessary for pzh.webinos.org and other deployments.

Jira issue: WP-396

Commit:
Update the PZH Web Interface to support issued SSL certificates.

The PZH Web Interface now supports an issued SSL certificate as well
as loading intermediate certificates.  It also uses a different key for
connecting to the PZH TLS Server (pzh.webinos.org:80) compared with hosting
an SSL web server (pzh.webinos.org:443).

Jira issue: WP-396

Commit:
Update PZH Web Server to support issued SSL certificates

(2) : Minor change to remove debug code.

Jira issue: WP-396

habibvirji commented on webinos/core/manager/certificate_manager/lib/certificate.js:
```diff
@@ -25,7 +25,7 @@ var Certificate = function () {
     this.cert.internal= {};
     this.cert.external= {};
-    this.cert.internal= {master: {}, conn: {}, web: {}};
+    this.cert.internal= {master: {}, conn: {}, webclient: {}, webssl: {}};
```
Would prefer if condition for webclient and webssl to add when it is PzhWS or PzhSSL

jplyle replied:
At this point in the code, there's no knowledge of whether it is a PZH, PZP or PZPWS invoking this function.

Where would I be putting the 'if' condition?  Sorry if this is a stupid question, I was just following the existing style of the code.

habibvirji replied:
Agreed.. you could just keep it master and conn here and in getSelfSignedCert, add if condition to add webclient and webssl in the internal structure.

TobyEalden commented:
Out of interest - will this prevent people from running a PZH if they don't have a properly signed cert?

jplyle commented:
Nope - it's just a few extra fields so that people can use a real certificate if they want to.  It'll all work as per normal otherwise.

Commit:
Improve certificate.js to selectively add webclient/webssl objects.

This is based on Habib's feedback:
https://github.com/webinos/Webinos-Platform/pull/447#discussion-diff-2828736

Jira issue: WP-396

jplyle commented:
Hi @habibvirji - does this work?
